src/main/java/com/st/utopia/agent/controller/SeatTickets.java

```diff
@@ -0,0 +1,189 @@
+
+import com.st.utopia.agent.model.PaymentAmount;
+import com.st.utopia.agent.model.SeatLocation;
+import com.st.utopia.agent.model.Ticket;
+import com.st.utopia.agent.model.User;;
+
+/**
+ * Controller for agents using this user-facing interface
+ *
+ * @author Jonathan Lovelace
+ * @author Salem O
+ */
+@RestController
+public class SeatTickets {
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@Value("${utopia.search.API}")
+	private String searchAPI;
+
+	@Value("${utopia.booking.API}")
+	private String bookingAPI;
+
+	@Value("${utopia.cancellation.API}")
+	private String cancellationAPI;
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for "get-all"
+	 * methods.
+	 * @param url the URL to send the REST request to
+	 * @param <T> the type we expect
+	 * @return the response the server sent
+	 */
+	private <T> ResponseEntity<T> methodCall(final String url, final HttpMethod method) {
+		return restTemplate.exchange(url, method, null,
+				new ParameterizedTypeReference<T>() {});
+	}
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for a method call.
+	 * @param url the URL to send the REST request to
+	 * @param type of method to send the REST request to
+	 * @param HttpEntity to send in the REST request
+	 * @param <T> the type we expect
+	 * @param <U> the type of the body sent to the server
+	 * @return the response the server sent
+	 */
+	private <T, U> ResponseEntity<T> methodCall(final String url, final HttpMethod method,
+			final HttpEntity<U> body) {
+		return restTemplate.exchange(url, method, body,
+				new ParameterizedTypeReference<T>() {});
+	}
+
+	/**
+	 * Get list of seats on a given flight
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seats")
+	public ResponseEntity<SeatLocation> getAllSeatsWithPlan(
+			@PathVariable final String flightId) {
+		// FIXME: Search service doesn't yet provide this
+		String url = searchAPI + "/seats?flight=" + flightId;
+		return this.<SeatLocation>methodCall(url, HttpMethod.GET);
+	}
+
+	/**
+	 * Get list of seats on a given flight
+	 * 
+	 * @return list of seats
+	 */
+	@GetMapping("/flight/{flightId}/seat/{row}/{seatId}")
+	public ResponseEntity<Ticket> getTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId) {
+		String url = bookingAPI + "/details/flights/" + flightId + "/rows/" +
+				row + "/seats/" + seatId;
+		return this.<Ticket>methodCall(url, HttpMethod.GET);
+	}
+
+	/**
+	 * Pay the price for ticket or extend the time
+	 * 
+	 * @return a ticket or an object
+	 */
+	@PutMapping("/flight/{flightId}/seat/{row}/{seatId}/ticket")
+	public ResponseEntity<Object> updateTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId, @RequestBody final PaymentAmount pay) {
+		// may need to just create a new object
+		if (pay.getPrice() == null) {
```

Comment:
Personally I would try to reduce the amount of duplication between the two branches of the `if`, but if that's too complicated for you that's fine.

Reply:
Yes, but it would be a little too complicated